backend/src/api/auth.ts

```diff
@@ -45,12 +45,18 @@ interface UserInfo {
 }
 
 const client = jwksClient({
-  jwksUri: `https://cognito-idp.us-east-1.amazonaws.com/${process.env.REACT_APP_USER_POOL_ID}/.well-known/jwks.json`
+  jwksUri: `https://cognito-idp.us-east-1.amazonaws.com/${process.env.REACT_APP_USER_POOL_ID}/.well-known/jwks.json`,
+  timeout: 30000 // Defaults to 30s
 });
+
 function getKey(header, callback) {
   client.getSigningKey(header.kid, function (err, key) {
-    const signingKey = key.getPublicKey();
-    callback(null, signingKey);
+    if (key) {
```

Comment:
We should just replace this with

```
    const signingKey = key.getPublicKey();
    callback(null, key ? signingKey: null);
```

Reply:
the warning ('key' is possibly 'undefined'.ts(18048)) is on this line  key.getPublicKey(); so this will not fix it. I can look to re-write like this if you agree:
    let signingKey;
    if (key) {
      signingKey = key.getPublicKey();
    }
    callback(null, signingKey);

Reply:
I think what you have now is fine then!

Reply:
Or what if you just did

```
    const signingKey = key?.getPublicKey();
    callback(null, signingKey);
```

Would this remove the warning?

Reply:
This works.